### DISCUSSION TITLE
tegra-demo-distro bitbake demo-image-full never completes

### DISCUSSION THREAD

**Discussion by pico-martin** [OE4T/tegra-demo-distro#282] [General]
Not sure if this is a discussion or an issue but will start here:

--> due to (at least) libnvidia-container-tools-0.10.0-r1 do_fetch
The fetch stage gets to between 21% (nearly always) to 36% (rarely) before hanging forever (until I kill it)

If I download using the same command but from the host context, it successfully downloads:
|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
martin@eol-dev:/dev/shm$ LANG=C git -c core.fsyncobjectfiles=0 -c gc.autoDetach=false -c core.pager=cat clone --bare --mirror https://github.com/NVIDIA/libnvidia-container.git /dev/shm/github.com.NVIDIA.libnvidia-container.git --progress
Cloning into bare repository '/dev/shm/github.com.NVIDIA.libnvidia-container.git'...
remote: Enumerating objects: 18810, done.
remote: Counting objects: 100% (1787/1787), done.
remote: Compressing objects: 100% (1355/1355), done.
remote: Total 18810 (delta 415), reused 1732 (delta 389), pack-reused 17023
Receiving objects: 100% (18810/18810), 2.04 GiB | 6.58 MiB/s, done.
Resolving deltas: 100% (6062/6062), done.
|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
However when I copy this to my downloads folder and try to continue, I get a raft of do_patch errors most likely due to and incomplete do_fetch stage...

Here is the output of my ever-hanging  'bitbake demo-image-full'
|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
martin@eol-dev:~/eol/tegra-demo-distro/tegrademo$ bitbake demo-image-full
Loading cache: 100% |####################################################################################################################################################################| Time: 0:00:00
Loaded 4302 entries from dependency cache.
Parsing recipes: 100% |##################################################################################################################################################################| Time: 0:00:00
Parsing of 2798 .bb files complete (2797 cached, 1 parsed). 4303 targets, 128 skipped, 0 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "2.0.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-oe4t-linux"
MACHINE              = "jetson-nano-devkit"
DISTRO               = "tegrademo"
DISTRO_VERSION       = "4.0.14+snapshot-f60fb520553cec2e8278d41988f4262813e30a30"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 = "HEAD:f60fb520553cec2e8278d41988f4262813e30a30"
meta-tegra           
contrib              = "HEAD:86c19e0d8b795898cf582e04bdf072af4e5b28d8"
meta-oe              
meta-python          
meta-networking      
meta-filesystems     = "HEAD:7da6cb848bc42b3e6bd5d2b37b52ba75510a6ca0"
meta-virtualization  = "HEAD:67b0ef4256a484851f3556d281ca4caa88374388"
meta-tegra-community = "HEAD:1cf7e5d8f1fa5b0d065f4f03ada43b9eedbb7d88"
meta-tegra-support   
meta-demo-ci         
meta-tegrademo       = "kirkstone-l4t-r32.7.x:cfe8920563cf0572fb6d8e603da1c3270947320f"

Initialising tasks: 100% |###############################################################################################################################################################| Time: 0:00:03
Sstate summary: Wanted 10 Local 1 Mirrors 0 Missed 9 Current 3754 (10% match, 99% complete)
NOTE: Executing Tasks
Setscene tasks: 3764 of 3764
Currently  1 running tasks (5850 of 8754)  66% |####################################################################################################                                                   |
0: libnvidia-container-tools-0.10.0-r1 do_fetch - 3m50s (pid 14408)  21% |##########################

**Comment by madisongh** [comment]
Is there anything in the `log.do_fetch` log file that might provide a clue?

Have you tried `bitbake -c cleanall libnvidia-container-tools` (to make sure everything related to that recipe gets wiped) followed by `bitbake -c fetch libnvidia-container-tools` (to try just the fetch step), to see if that works OK, instead of manually cloning and copying?

What is your host environment?

**Reply by pico-martin** [reply]
Thank you Madisongh, I tried (nearly) everything including your suggestions above to no avail. I then started from scratch again with an Ubuntu 22.04 (previously Ubuntu 18.04 which I (mis)understood to be the recommended host OS for kirkstone 32.7.4) - the build ran from start to finish.
What mislead me for so long was its failure on this specific recipe.

**Reply by madisongh** [reply]
Yeah, 18.04 is pretty old now, although I'm not sure why that particular step would fail on that version.  If it's working with 22.04, great.
